doc/src/Examples.dox

```diff
@@ -77,20 +77,26 @@
  * >    }
  * > }
  *
- * # We must first customize the settings for the specific manager.
- * # The available settings are defined by the manager implementation.
- * # The settings provided by availableManagers will be the complete
- * # set of available settings with their default values.
- * settings = availableManagers['org.openassetio.example.manager'].settings
- * settings["server"] = "my.server.com"
- *
- * # We then construct the Manager that we'd like to use. Note that the
- * # manager is constructed in an uninitialized state and cannot be used
- * # immediately (except for certain metadata queries).
+ * # Once we know which manager we wish to use, we can ask the factory
+ * # to create one for us.
  * manager = managerFactory.createManager('org.openassetio.example.manager')
  *
- * # Finally, we can initialize the manager with our settings. This can
- * # be called again in order to change settings, if required.
+ * # We now have an instance of the requested manager, but it is not
+ * # quite ready for use yet. The manager returned by the
+ * # ManagerFactory needs to be initialized before it can be used to
+ * # query or publish assets. Setup is split into two stages to allow
+ * # adjustments to its settings to be made prior to use if required.
+ *
+ * # A manager's current (or in this case default) settings can be
+ * # queried if needed:
+ * settings = manager.settings()
+ * # ...and updated with new values as desired.
+ * settings["server"] = "my.server.com"
+ *
+ * # Finally, we can initialize the manager with the desired settings,
+ * # preparing it for use. Note that this may include non-trivial
+ * # amounts of work. Settings updates are sparse, so if you don't have
+ * # any custom settings, you can pass an empty dictionary here.
```

Comment:
[question] Should we make it so the `Manager` wrapper makes settings optional?

Reply:
:thinking: Forcing the user to pass settings has a couple of advantages
* Passing an empty dict explicitly states that they want the default settings and must have considered the consequences.
* Throwing an error if they don't pass any settings acts as a form of documentation, in case the user forgets when and how to set settings.
* Other than the first `initialize` call, it doesn't make sense to call `initialize` without (a change of) settings(?).
* If we start strict we can make it optional later, but not vice versa.

Having said all that, I don't have a strong opinion if we want to make it optional.  Especially if it turns out that default settings are the most common case.

Reply:
Im sold :) stick as we are!